### ISSUE TITLE
Rename file before upload

### ISSUE THREAD

**Issue by NotMyUid** [fortana-co/react-dropzone-uploader#176]
Hello,
I'm trying to rename a file before uploading it. Is it possible?

I tried renaming "file.name" both in "getUploadParams" and in "onChangeStatus" function, but I noticed that it is a readonly field.

Code:
![1](https://user-images.githubusercontent.com/50867282/140778105-342e833a-8b63-48db-b471-550c379bfa81.PNG)

I also tried, in "getUploadParams", to modify the returned value including a new File with the modified name as a property of "fields" object: in this case, in the request there are two files instead of one.

Code: 
![2](https://user-images.githubusercontent.com/50867282/140778282-ecff72ce-d70c-4acf-8898-6bd26d3eb607.PNG)

Request:
![3](https://user-images.githubusercontent.com/50867282/140779864-c512a65c-1243-4370-b6b1-a219c90bdee2.PNG)

Thank you so much!

**Comment by mariomurrent-softwaresolutions** [comment]
@NotMyUid Did you solve this?

**Comment by NotMyUid** [comment]
Hi @mariomurrent-softwaresolutions ,
Solved with the use of a work-around.. I post here the steps we followed in order to rename a file before it has been uploaded.

1. Step one consists in disabling auto-upload (no-longer use "getUploadParams").
2. Keep listening for status change, then create a copy of the original file, rename it, and store in a global variable.

Code:
![0](https://user-images.githubusercontent.com/50867282/205992585-6100a96f-6f9a-44c5-bca1-b80e3cdc0bb1.PNG)


3. Create a method which uploads the new file to your destination.

Code:
![1](https://user-images.githubusercontent.com/50867282/205992622-9e0482d1-a008-4055-8f38-8050c1a0f266.PNG)
![2](https://user-images.githubusercontent.com/50867282/205992644-97ac6b32-bdc2-4486-a617-8d8d42eed345.PNG)


4. Add a button to your template in order to upload the image, on click event call the method you wrote at step 3.


Hope this helps, best regards :)

**Comment by mariomurrent-softwaresolutions** [comment]
@NotMyUid Yeah I found the same stuff. Thx for information. This works like a charm although its a bit hacky
